suggest/suggest.go

```diff
@@ -72,7 +73,8 @@ func (e *Engine) loop() {
 			select {
 			case toBeAdded := <-e.incomingCh:
 				e._add(toBeAdded)
-			default:
+				// Wait a little to declare adds done to avoid a tight loop
+			case <-time.After(100 * time.Millisecond):
```

Comment:
It seems to me that each time you enter the select statement it will create a new timer which will go out of scope and then be later garbage collected. This will add load to the garbage collector and may also slow down the runtime, which manages the timers.
Take a look at the idleMarkTimer in the Symantec/Dominator/lib/logbuf for another way of doing things.